src/buildHTML.js

```diff
@@ -79,14 +78,25 @@ export const buildExpression = function(expression, options, isRealGroup) {
     const nonSpaces =
         rawGroups.filter(group => group && group.classes[0] !== "mspace");
 
+    // add dummy spans for determining spacings between surrounding atoms
+    nonSpaces.unshift(surrounding[0] &&
+        new makeSpan([surrounding[0]], [], options));
+    nonSpaces.push(surrounding[1] &&
+        new makeSpan([surrounding[1]], [], options));
```

Comment:
Is it necessary to create the spans if `surrounding[0]` or `surrounding[1]` is `null` or `undefined`?

Reply:
@kevinbarabash It's for easy manipulation of `nonSpaces`. This way, we can safely assume that `nonSpaces[0]` and `nonSpaces[nonSpaces.length -1]` are dummy spans.

Reply:
I would rewrite this as:
```
const nonSpaces = [
   surrounding[0] && makeSpan([surrounding[0]], [], options)),
   rawGroups.filter(group => group && group.classes[0] !== "mspace"),
   surrounding[1] && makeSpan([surrounding[1]], [], options)),
];
```
This avoids the `unshift` which could be costly depending on the length of the array and better shows that the `surrounding` array is being used to surround something.

Note: `new` isn't required for `makeSpan`.

Reply:
I can't wait until we have everything converted to `flow` so that we can better enforce some of these assumptions.